### PR TITLE
Add Cosmote TV

### PR DESCRIPTION
**Pull request by down-bad**
Cosmote TV is a Greek pay television service. This presence should support all functionality of the website, such as watching live TV, replay TV, streaming movies, series, and all the pages of the site.

Home page
Screenshot 2021-09-11 022045
Live TV
Screenshot 2021-09-11 022340
Replay TV
Screenshot 2021-09-11 022501
Movie/Series/etc
Screenshot 2021-09-11 022747

### CODE REVIEW

**Comment by down-bad** [comment]
Changes image based on the channel (supports all available channels)

Examples:
Screenshot 2021-09-11 144757
Screenshot 2021-09-11 144614

File: websites/C/Cosmote TV/presence.ts [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    if (live === false) {
+      (data.smallImageKey = paused ? "pause" : "play"),
+        (data.smallImageText = paused
+          ? (await strings).pause
+          : (await strings).play);
+
+      (data.startTimestamp = timestamps[0]),
+        (data.endTimestamp = timestamps[1]);
+
+      var series = document.querySelector(
```
Comment (EncryptedDev) [inline websites/C/Cosmote TV/presence.ts:102]:
```suggestion
      const series = document.querySelector(
```

File: websites/C/Cosmote TV/presence.ts [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        ".meta-title[ng-bind='details.title']"
+      )
+      ?.textContent,
+
+     video: HTMLVideoElement = document.querySelector("video#arxPlayer"),
+
+     {paused} = video,
+     {currentTime} = video,
+     {duration} = video,
+     timestamps = getTimestamps(Math.floor(currentTime), Math.floor(duration)),
```
Comment (EncryptedDev) [inline websites/C/Cosmote TV/presence.ts:85]:
```suggestion
     timestamps = presence.getTimestamps(Math.floor(currentTime), Math.floor(duration)),
```

File: websites/C/Cosmote TV/presence.ts [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+      .querySelector(
+        ".meta-title[ng-bind='details.title']"
+      )
+      ?.textContent,
+
+     video: HTMLVideoElement = document.querySelector("video#arxPlayer"),
+
+     {paused} = video,
+     {currentTime} = video,
+     {duration} = video,
```
Comment (EncryptedDev) [inline websites/C/Cosmote TV/presence.ts:84]:
```suggestion
     { paused, currentTime, duration } = video,
```

File: websites/C/Cosmote TV/presence.ts [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+  });
+
+function getTimestamps(
+  videoTime: number,
+  videoDuration: number
+): Array<number> {
+  const startTime = Date.now(),
+   endTime = Math.floor(startTime / 1000) - videoTime + videoDuration;
+  return [Math.floor(startTime / 1000), endTime];
+}
```
Comment (EncryptedDev) [inline websites/C/Cosmote TV/presence.ts:18]:
```suggestion
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/C/Cosmote TV/presence.ts [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+  clientId: "883446187099840562"
+}),
+  strings = presence.getStrings({
+    play: "general.playing",
+    pause: "general.paused",
+    live: "general.live",
+    search: "general.searchFor"
+  });
+
+
```
Comment (EncryptedDev) [inline websites/C/Cosmote TV/presence.ts:11]:
```suggestion
```

File: websites/C/Cosmote TV/presence.ts [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+  else if (document.location.href.includes("onDemand/DOCUMENTARIES"))
+    data.details = "Browsing Documentaries...";
+  else if (document.location.href.includes("onDemand/KIDS"))
+    data.details = "Browsing Kids content...";
+  else if (document.location.href.includes("watchlist"))
+    data.details = "Viewing watchlist...";
+  else
+    data.details = "Browsing...";
+
+  data.startTimestamp = Math.floor(Date.now() / 1000);
```
Comment (EncryptedDev) [inline websites/C/Cosmote TV/presence.ts:51]:
```suggestion
```

File: websites/C/Cosmote TV/presence.ts [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+    play: "general.playing",
+    pause: "general.paused",
+    live: "general.live",
+    search: "general.searchFor"
+  });
+
+
+presence.on("UpdateData", async () => {
+  const data: PresenceData = {
+    largeImageKey: "tv"
```
Comment (EncryptedDev) [inline websites/C/Cosmote TV/presence.ts:14]:
```suggestion
    largeImageKey: "tv",
    startTimestamp = Math.floor(Date.now() / 1000)
```

File: websites/C/Cosmote TV/presence.ts [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+    ? true
+    : false;
+
+  if (playerCheck) {
+    const title = document
+      .querySelector(
+        ".meta-title[ng-bind='details.title']"
+      )
+      ?.textContent,
+
```
Comment (EncryptedDev) [inline websites/C/Cosmote TV/presence.ts:65]:
```suggestion
```

File: websites/C/Cosmote TV/presence.ts [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+      video: HTMLVideoElement = document.querySelector("video#arxPlayer"),
+
```
Comment (EncryptedDev) [inline websites/C/Cosmote TV/presence.ts:67]:
```suggestion
```

File: websites/C/Cosmote TV/presence.ts [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+      { paused, currentTime, duration } = video,
+      timestamps = presence.getTimestamps(Math.floor(currentTime), Math.floor(duration)),
+
+      live = document.querySelector(
+        ".meta-remain"
+      )
+        ? true
+        : false;
+
+    if (live === false) {
```
Comment (EncryptedDev) [inline websites/C/Cosmote TV/presence.ts:77]:
```suggestion
    if (!live) {
```

File: websites/C/Cosmote TV/presence.ts [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+      .querySelector(
+        ".meta-title[ng-bind='details.title']"
+      )
+      ?.textContent,
+
+      video: HTMLVideoElement = document.querySelector("video#arxPlayer"),
+
+      { paused, currentTime, duration } = video,
+      timestamps = presence.getTimestamps(Math.floor(currentTime), Math.floor(duration)),
+
```
Comment (EncryptedDev) [inline websites/C/Cosmote TV/presence.ts:70]:
```suggestion
```

File: websites/C/Cosmote TV/presence.ts [hunk 12]
```diff
@@ -0,0 +1,135 @@
...
+      data.details = title;
+    }
+
+    const channel = document
+      .querySelector(
+        ".meta-title[ng-bind='details.channel.title']"
+      )
+      ?.textContent;
+
+    if (channel !== '') {
```
Comment (EncryptedDev) [inline websites/C/Cosmote TV/presence.ts:116]:
```suggestion
    if (channel.length === 0) {
```

File: websites/C/Cosmote TV/presence.ts [hunk 13]
```diff
@@ -0,0 +1,135 @@
...
+      if (live === false)
```
Comment (EncryptedDev) [inline websites/C/Cosmote TV/presence.ts:117]:
```suggestion
      if (!live)
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

**Comment by EncryptedDev** [comment]
Upload further proof of working please

**Comment by down-bad** [comment]
Home
Screenshot 2021-09-12 122105
Screenshot 2021-09-12 122132

Channels
Screenshot 2021-09-12 122154
Screenshot 2021-09-12 123623

Series
Screenshot 2021-09-12 123206
Screenshot 2021-09-12 123219
etc...

Live TV
Screenshot 2021-09-12 122344
Screenshot 2021-09-12 123558

Replay TV
Screenshot 2021-09-12 122523
Screenshot 2021-09-12 122538

Movie/Series/other
Screenshot 2021-09-12 123117
Screenshot 2021-09-12 123043

Paused
Screenshot 2021-09-12 123954
Screenshot 2021-09-12 124009

**Review by EncryptedDev — DISMISSED** [review]
Seems cool with me, await proper approval from maintainers shortly.

File: websites/C/Cosmote TV/presence.ts [hunk 14]
```diff
@@ -0,0 +1,130 @@
...
+
+presence.on("UpdateData", async () => {
+  const data: PresenceData = {
+    largeImageKey: "tv",
+    startTimestamp: Math.floor(Date.now() / 1000)
+  };
+  if (document.location.href.includes("search")) {
+    data.details = "Searching...";
+    data.smallImageKey = "search";
+    data.smallImageText = await (await strings).search;
```
Comment (Slowlife01) [inline websites/C/Cosmote TV/presence.ts:19]:
```suggestion
    data.smallImageText = (await strings).search;
```

File: websites/C/Cosmote TV/presence.ts [hunk 15]
```diff
@@ -0,0 +1,130 @@
...
+    pause: "general.paused",
+    live: "general.live",
+    search: "general.searchFor"
+  });
+
+presence.on("UpdateData", async () => {
+  const data: PresenceData = {
+    largeImageKey: "tv",
+    startTimestamp: Math.floor(Date.now() / 1000)
+  };
```
Comment (Slowlife01) [inline websites/C/Cosmote TV/presence.ts:15]:
```suggestion
  }),
  startedTime = Math.floor(Date.now() / 1000);

presence.on("UpdateData", async () => {
  const data: PresenceData = {
    largeImageKey: "tv",
    startTimestamp: startedTime
  };
```

File: websites/C/Cosmote TV/presence.ts [hunk 16]
```diff
@@ -0,0 +1,130 @@
...
+        ".meta-remain"
+      )
+        ? true
+        : false;
+
+    if (!live) {
+      (data.smallImageKey = paused ? "pause" : "play"),
+        (data.smallImageText = paused
+          ? (await strings).pause
+          : (await strings).play);
```
Comment (Slowlife01) [inline websites/C/Cosmote TV/presence.ts:76]:
```suggestion
      data.smallImageKey = paused ? "pause" : "play";
      data.smallImageText = paused
          ? (await strings).pause
          : (await strings).play;
```

File: websites/C/Cosmote TV/presence.ts [hunk 17]
```diff
@@ -0,0 +1,130 @@
...
+
+      (data.startTimestamp = timestamps[0]),
+        (data.endTimestamp = timestamps[1]);
```
Comment (Slowlife01) [inline websites/C/Cosmote TV/presence.ts:79]:
```suggestion
      data.endTimestamp = timestamps.pop();
```

File: websites/C/Cosmote TV/presence.ts [hunk 18]
```diff
@@ -0,0 +1,130 @@
...
+          "span[ng-bind='details.seriesSubs']"
+        ).textContent;
+
+      } else
+        data.details = title;
+    } else {
+      (data.smallImageKey = paused ? "pause" : "live"),
+        (data.smallImageText = paused
+          ? (await strings).pause
+          : (await strings).live);
```
Comment (Slowlife01) [inline websites/C/Cosmote TV/presence.ts:98]:
```suggestion
      data.smallImageKey = paused ? "pause" : "live";
      data.smallImageText = paused
          ? (await strings).pause
          : (await strings).live;
```

File: websites/C/Cosmote TV/presence.ts [hunk 19]
```diff
@@ -0,0 +1,130 @@
...
+      data.details = title;
+    }
+
+    const channel = document
+      .querySelector(
+        ".meta-title[ng-bind='details.channel.title']"
+      )
+      ?.textContent;
+
+    if (channel.length !== 0) {
```
Comment (Slowlife01) [inline websites/C/Cosmote TV/presence.ts:111]:
```suggestion
    if (channel?.length !== 0) {
```

**Review by Slowlife01 — CHANGES_REQUESTED** [review]

**Review by Slowlife01 — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**